From 6a64ded60230c818ee88f3183358d7c9c12d5eb5 Mon Sep 17 00:00:00 2001
From: danieleTrimarchi <daniele.trimarchi@iconeus.com>
Date: Sun, 22 Nov 2020 15:18:36 +0100
Subject: [PATCH] proposal to resolve issue #41

---
 synology_api/filestation.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/synology_api/filestation.py b/synology_api/filestation.py
index a8d50b9..a732a25 100644
--- a/synology_api/filestation.py
+++ b/synology_api/filestation.py
@@ -980,7 +980,7 @@ class FileStation:
 
         return self.request_data(api_name, api_path, req_param)
 
-    def get_file(self, path=None, mode=None, chunkSize=8192):
+    def get_file(self, path=None, mode=None, chunkSize=8192, dest_path="."):
         api_name = 'SYNO.FileStation.Download'
         info = self.file_station_list[api_name]
         api_path = info['path']
@@ -1006,7 +1006,9 @@ class FileStation:
         if mode == r'download':
             with session.get(url, stream=True) as r:
                 r.raise_for_status()
-                with open(os.path.basename(path), 'wb') as f:
+                if not os.path.isdir(dest_path):
+                    os.makedirs(dest_path)
+                with open(dest_path + "/" +os.path.basename(path), 'wb') as f:
                     for chunk in r.iter_content(chunk_size=chunkSize):
                         if chunk:  # filter out keep-alive new chunks
                             f.write(chunk)
-- 
2.24.1.windows.2

